### ISSUE TITLE
Images in slideshow are not ordered if added by folder (random setting is disabled)

### ISSUE THREAD

**Issue by ximex** [obsproject/obs-studio#10382]
### Operating System Info

Other

### Other OS

Kubuntu 23.10

### OBS Studio Version

30.1.0

### OBS Studio Log URL

https://obsproject.com/logs/A8fSnIT7ryAcMhXK

### Expected Behavior

If i add a folder for a slideshow they should be in alphabetically order of the filename if the random flag is not enabled

### Current Behavior

The images appear in a random order (the random flag is NOT enabled)

### Steps to Reproduce

1. Add a slidshow
2. Add a dir with many images in it
3. Compare the ordering in the folder with the order the images are shown

### Anything else we should know?

I'm not the `C` expert but if i'm not wrong it's because the `os_readdir` function doesn't sort the entries in a directory as the function `readdir` doesn't guaranteed the ordering AFAIK.
See:
https://github.com/obsproject/obs-studio/blob/5f2bb795f07b3174c991b7bd2926460bc0c33a39/plugins/image-source/obs-slideshow-mk2.c#L500
https://github.com/obsproject/obs-studio/blob/5f2bb795f07b3174c991b7bd2926460bc0c33a39/libobs/util/platform-nix.c#L539
https://github.com/obsproject/obs-studio/blob/5f2bb795f07b3174c991b7bd2926460bc0c33a39/libobs/util/platform-nix.c#L546

**Comment by RytoEX** [comment]
As far as I can tell, the old Image Slideshow in OBS Studio 30.0.2 and older uses the same function calls, so the behavior should also occur there and anywhere we call `os_readdir` for a list of files on Linux or macOS.

**Comment by RytoEX** [comment]
While unusual, I don't strictly think this is a bug. As you said, `readdir` doesn't guarantee a file order, and we do not sort the received list of files, so this is currently working as designed.

**Comment by ximex** [comment]
I don't think it is as designed. as then the random option would be useless. and yes what i saw it's also in the old slideshow. I'm not sure about `readdir` behavior. i only read about it on the internet

**Comment by RytoEX** [comment]
> I don't think it is as designed. as then the random option would be useless. and yes what i saw it's also in the old slideshow. I'm not sure about `readdir` behavior. i only read about it on the internet

The man page for `readdir` says:
> The order in which filenames are read by successive calls to readdir() depends on the filesystem implementation; it is unlikely that the names will be sorted in any fashion.

We do not sort the filenames from `readdir`. Thus, the code itself is working _as written_, which is what I mean by "working as designed". The "random" option affects the list of files that the user has added, which is much more apparent when you have added files manually.

**Comment by ximex** [comment]
But could that get implemented that the files get sorted by filename by default?

**Comment by RytoEX** [comment]
> But could that get implemented that the files get sorted by filename by default?

It's technically possible, but that would be a feature request. We use GitHub Issues exclusively for bug reports. Feature requests 
are tracked on our [forums](https://obsproject.com/forum/), our [Ideas and Suggestions site](https://ideas.obsproject.com/), or the \#ideas-suggestions channel in our [Discord](https://obsproject.com/discord/).
